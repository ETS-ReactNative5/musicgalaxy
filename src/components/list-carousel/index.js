export { ListCarousal } from './list-carousel';