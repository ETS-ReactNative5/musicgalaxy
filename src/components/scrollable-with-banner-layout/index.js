export { ScrollableWithBannerLayout } from './scrollable-with-banner-layout';
