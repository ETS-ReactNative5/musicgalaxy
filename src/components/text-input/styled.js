import styled from 'styled-components';
import { APP_COLORS } from '@src/theme/colors';
import { scale } from '@src/utils/media';
import { radius, spacing } from '@src/theme/spacing';
import { TextInput } from 'react-native';


export const StyledTextInput = styled(TextInput)`
  border: ${APP_COLORS.white};
  padding: ${spacing.small}px;
  padding-left: ${({ leftIconPadding }) =>
    scale(leftIconPadding) || spacing.small}}px;
  color: ${APP_COLORS.white};
  border-radius: ${radius.default}
  font-size: ${scale(16)};
`;

