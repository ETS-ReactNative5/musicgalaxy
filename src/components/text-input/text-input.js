import { APP_COLORS } from '@src/theme/colors';
import { TEST_IDS } from '@src/utils/constants';
import React from 'react';
import { View } from 'react-native';

import { StyledTextInput } from './styled';
/**
 * Generic Text Input component to render the Custom Input with Icon in it
 */
export const TextInput = ({ LeftIcon, ...props }) => {
  return (
    <View>
      {LeftIcon}
      <StyledTextInput
        testID={TEST_IDS.TEXT_INPUT}
        placeholderTextColor={APP_COLORS.red}
        {...props}
      />
    </View>
  );
};
