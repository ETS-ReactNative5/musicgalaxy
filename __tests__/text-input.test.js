import React from 'react';

import { render, fireEvent } from '@testing-library/react-native';
import { TextInput } from '@src/components/text-input';

const sampleProps = {
    onChangeText: jest.fn(),
    value: '',
    placeholder: "Search by Artist or Song Title",
    onSubmitEditing: jest.fn()
}

describe('Testing TextInput Component', () => {

    it('should match the TextInput component snapshot', () => {
        const { toJSON } = render(<TextInput />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render TextInput placeholder correctly', () => {
        const { getByPlaceholderText } = render(<TextInput {...sampleProps} />);
        expect(getByPlaceholderText(sampleProps.placeholder)).not.toBeNull();
    });

    it('should update the placeholder Text correctly', () => {
        const { getByPlaceholderText } = render(<TextInput {...sampleProps} />);
        fireEvent.changeText(getByPlaceholderText(sampleProps.placeholder), 'John');
    });

    it('should call onChange Function correctly', () => {
        const { getByPlaceholderText } = render(<TextInput {...sampleProps} />);
        fireEvent(getByPlaceholderText(sampleProps.placeholder), 'onChangeText');
        expect(sampleProps.onChangeText).toHaveBeenCalled();
    });
    
});

