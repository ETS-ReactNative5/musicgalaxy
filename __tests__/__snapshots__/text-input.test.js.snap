// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextInput Component should match the TextInput component snapshot 1`] = `
<View>
  <TextInput
    className="sc-bczRLJ OLuDo"
    placeholderTextColor="red"
    testID="TEXT_INPUT"
  />
</View>
`;
